lib/dry/view/part.rb

```diff
@@ -21,11 +30,14 @@ class Part
 
       attr_reader :_renderer
 
-      def initialize(name:, value:, renderer: MissingRenderer.new, context: nil)
+      attr_reader :_decorated_childs
+
+      def initialize(name:, value:, renderer: MissingRenderer.new, context: nil, decorated_childs: {})
         @_name = name
         @_value = value
         @_context = context
         @_renderer = renderer
+        @_decorated_childs = decorated_childs
```

Comment:
Another solution I explore was that once you initialize the `Part` class it will recursively decorate the children something in the lines of
`@_decorated_childs = self.class.decorates.call(self)`